app/move/controllers/create/base.test.js

```diff
@@ -0,0 +1,127 @@
+const FormController = require('hmpo-form-wizard').Controller
+
+const CreateBaseController = require('./base')
+
+const controller = new CreateBaseController({ route: '/' })
+
+describe('Move controllers', function() {
+  describe('Create base controller', function() {
+    describe('#middlewareChecks()', function() {
+      beforeEach(function() {
+        sinon.stub(FormController.prototype, 'middlewareChecks')
+        sinon.stub(controller, 'use')
+
+        controller.middlewareChecks()
+      })
+
+      it('should call parent method', function() {
+        expect(FormController.prototype.middlewareChecks).to.have.been
+          .calledOnce
+      })
+
+      it('should call check current location method', function() {
+        expect(controller.use).to.have.been.calledWith(
+          controller.checkCurrentLocation
+        )
+      })
+    })
+
+    describe('#middlewareLocals()', function() {
+      beforeEach(function() {
+        sinon.stub(FormController.prototype, 'middlewareLocals')
+        sinon.stub(controller, 'use')
+
+        controller.middlewareLocals()
+      })
+
+      it('should call parent method', function() {
+        expect(FormController.prototype.middlewareLocals).to.have.been
+          .calledOnce
+      })
+
+      it('should call set cancel url method', function() {
+        expect(controller.use).to.have.been.calledWith(controller.setCancelUrl)
+      })
+    })
+
+    describe('#setCancelUrl()', function() {
+      let res, nextSpy
+
+      beforeEach(function() {
+        nextSpy = sinon.spy()
+        res = {
+          locals: {},
+        }
+      })
+
+      context('with no moves url local', function() {
+        beforeEach(function() {
+          controller.setCancelUrl({}, res, nextSpy)
+        })
+
+        it('should set cancel url correctly', function() {
+          expect(res.locals).to.have.property('cancelUrl')
+          expect(res.locals.cancelUrl).to.equal(undefined)
+        })
+
+        it('should call next', function() {
+          expect(nextSpy).to.be.calledOnceWithExactly()
+        })
+      })
+
+      context('with moves url local', function() {
+        beforeEach(function() {
+          res.locals.MOVES_URL = '/moves?move-date=2019-10-10'
```

Comment:
`'/moves?move-date=2019-10-10'` could go into a `const` here as its used a few times and giving a `const` a name may explain what it is a little more (if needed)

Reply:
Might leave this one as it only appears twice. I think this logic will be revisited again soon too when we start introducing more complex search to the dashboard.